workflow-bot-app/src/test/java/com/symphony/bdk/workflow/converter/DefaultObjectConverterTest.java

```diff
@@ -45,6 +45,18 @@ void convert_SourceToTarget_Succeed() {
     then(target).isNotNull();
   }
 
+  @Test
+  void convert_SourceToTarget_SucceedWithSpecificSourceClass() {
+    Target target = objectConverter.convert(new Source(), Source.class, Target.class);
+    then(target).isNotNull();
+  }
+
+  @Test
+  void convert_SourceToTarget_FailWithSpecificSourceClass() {
+    assertThatThrownBy(() -> objectConverter.convert(new Source(), Target.class, Target.class)).isInstanceOf(
```

Comment:
To check the exception message too, maybe we can use: 
`assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(
            () -> objectConverter.convert(new Source(), Target.class, Target.class))
        .satisfies(e -> assertThat(e.getMessage()).isEqualTo(.....));`